File: lab-2.1-ByLiza/С++/Line.cpp
#include "Line.h"
#include <cmath>

Line::Line(double startX, double startY, double endX, double endY)
    : startX(startX), startY(startY), endX(endX), endY(endY) {}

Line::Line() : startX(0), startY(0), endX(1), endY(1) {}

Line::Line(const Line& other)
    : startX(other.startX), startY(other.startY), endX(other.endX), endY(other.endY) {}

double Line::GetStartX(){ return startX; }
double Line::GetStartY(){ return startY; }
double Line::GetEndX(){ return endX; }
double Line::GetEndY(){ return endY; }

double Line::GetLength(){
    return sqrt(pow(endX - startX, 2) + pow(endY - startY, 2));
}

void Line::SetEndCoordinates(double newEndX, double newEndY) {
    endX = newEndX;
    endY = newEndY;
}

File: lab-2.1-ByLiza/LineApp/Line.cs
using System;

namespace LineApp
{
    public class Line
    {
        private double _startX, _startY, _endX, _endY;

        public Line(double startX, double startY, double endX, double endY)
        {
            _startX = startX;
            _startY = startY;
            _endX = endX;
            _endY = endY;
        }
        public Line()
        {
            _startX = _startY = 0;
            _endX = _endY = 1;
        }
        public Line(Line other)
        {
            _startX = other._startX;
            _startY = other._startY;
            _endX = other._endX;
            _endY = other._endY;
        }

        public double StartX { get => _startX; }
        public double StartY { get => _startY; }
        public double EndX { get => _endX; }
        public double EndY { get => _endY; }

        public double GetLength()
        {
            return Math.Sqrt(Math.Pow(_endX - _startX, 2) + Math.Pow(_endY - _startY, 2));
        }
        public void SetEndCoordinates(double newEndX, double newEndY)
        {
            _endX = newEndX;
            _endY = newEndY;
        }
    }
}